spec/factories/test_result.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :v2_test_result, class: 'V2::TestResult' do
+    tailoring { V2::Tailoring.find_by!(policy_id: policy_id, os_minor_version: system.os_minor_version) }
```

Comment:
This needs to be conditional, if the `policy_id` is set. Otherwise it should create an association.